src/test/java/src/test/java/ru/training/at/hw2/ex1/SoftAssertsTest.java

```diff
@@ -0,0 +1,163 @@
+        // Create new instance of WebDriver
+        webDriver = new ChromeDriver();
+        // implicit waits
+        webDriver.manage().timeouts().implicitlyWait(2, TimeUnit.SECONDS);
+        // Start browser maximized
+        webDriver.manage().window().maximize();
+
+        // 1. Open test site by URL
+        String baseUrl = "https://jdi-testing.github.io/jdi-light/index.html";
+        webDriver.navigate().to(baseUrl);
+
+        // 2. Assert Browser title
+        softAssert.assertEquals(webDriver.getTitle(), "Home Page");
+
+        // 3. Perform login
+        WebElement loginIcon = webDriver.findElement(By.id("user-icon"));
+        loginIcon.click();
+        WebElement usernameInput = webDriver.findElement(By.id("name"));
+        usernameInput.sendKeys("Roman");
+        WebElement passwordInput = webDriver.findElement(By.id("password"));
+        passwordInput.sendKeys("Jdi1234");
+        WebElement loginButton = webDriver.findElement(By.id("login-button"));
+        loginButton.click();
+
+        // 4. Assert Username is logged in
+        String userName = webDriver.findElement(By.id("user-name")).getText();
+        softAssert.assertEquals(userName, "ROMAN IOVLEV");
+
+        // 5. Assert that there are 4 items on the header section are displayed
+        // and they have proper texts
+        String navString = ".uui-navigation.nav.navbar-nav.m-l8 > li";
+        List<WebElement> navElements = webDriver.findElements(By.cssSelector(navString));
+        List<String> navElementsStringList = new ArrayList<>();
+        // Get text from nav WebElement, count elements, check isDisplayed()
+        int navStringCount = 0;
+        boolean isDisplayedNav = true;
+        for (WebElement web : navElements) {
+            // do not put empty Strings into List
+            System.out.println(web.getText());
+            if (web.getText().length() > 0) {
+                navElementsStringList.add(web.getText());
+                navStringCount++;
+            }
+            if (!web.isDisplayed()) {
+                isDisplayedNav = false;
+            }
+        }
+        softAssert.assertTrue(navElementsStringList.contains("HOME"));
+        softAssert.assertTrue(navElementsStringList.contains("CONTACT FORM"));
+        softAssert.assertTrue(navElementsStringList.contains("SERVICE"));
+        softAssert.assertTrue(navElementsStringList.contains("METALS & COLORS"));
+        softAssert.assertEquals(navStringCount, 4);
+        softAssert.assertTrue(isDisplayedNav);
+
+        // 6. Assert that there are 4 images on the Index Page and they are displayed
+        WebElement iconPractise = webDriver.findElement(By.cssSelector(".icon-practise"));
+        WebElement iconCustom = webDriver.findElement(By.cssSelector(".icon-custom"));
+        WebElement iconMulti = webDriver.findElement(By.cssSelector(".icon-multi"));
+        WebElement iconBase = webDriver.findElement(By.cssSelector(".icon-base"));
```

Comment:
icons could be find as `List<webElement>` and through the loop could be checked for the `isDisplayer`

Reply:
Icons common parent (ancestor) is too far away to get them by findElements(). The code will be too cumbersome. And I don't know how to check whether I got the right icons in my List<WebElement>. For String Lists, I can simply print them with println().